ipa-core/src/protocol/ipa_prf/shuffle/mod.rs

```diff
@@ -111,17 +145,90 @@ where
     }
 }
 
+// This function converts Attribution Outputs to an AdditiveShare needed for shuffle protocol
+pub fn attribution_outputs_to_shuffle_input<BK, TV, YS>(
+    input: &SecretSharedAttributionOutputs<BK, TV>,
+) -> AdditiveShare<YS>
+where
+    YS: BooleanArray,
+    BK: BooleanArray,
+    TV: BooleanArray,
+{
+    let mut y = AdditiveShare::new(YS::ZERO, YS::ZERO);
+    expand_shared_array_in_place(&mut y, &input.attributed_breakdown_key_bits, 0);
+    expand_shared_array_in_place(
+        &mut y,
+        &input.capped_attributed_trigger_value,
+        BK::BITS as usize,
+    );
+    y
+}
+
+// This function converts Attribution Outputs  obtained from shuffle protocol to OprfReport
+pub fn shuffled_to_attribution_outputs<YS, BK, TV>(
+    input: &AdditiveShare<YS>,
+) -> SecretSharedAttributionOutputs<BK, TV>
+where
+    YS: BooleanArray,
+    BK: BooleanArray,
+    TV: BooleanArray,
+{
+    let attributed_breakdown_key_bits = extract_from_shared_array::<YS, BK>(input, 0);
+    let capped_attributed_trigger_value =
+        extract_from_shared_array::<YS, TV>(input, BK::BITS as usize);
+
+    SecretSharedAttributionOutputs {
+        attributed_breakdown_key_bits,
+        capped_attributed_trigger_value,
+    }
+}
+
 #[cfg(all(test, unit_test))]
 pub mod tests {
     use rand::Rng;
 
     use crate::{
-        ff::boolean_array::{BA20, BA3, BA8},
-        protocol::ipa_prf::shuffle::shuffle_inputs,
+        ff::{
+            boolean::Boolean,
+            boolean_array::{BA20, BA3, BA32, BA64, BA8},
+            U128Conversions,
+        },
+        protocol::{
+            context::UpgradedSemiHonestContext,
+            ipa_prf::{
+                prf_sharding::{
+                    tests::PreAggregationTestOutputInDecimal, AttributionOutputsTestInput,
+                    SecretSharedAttributionOutputs,
+                },
+                shuffle::{shuffle_attribution_outputs, shuffle_inputs},
+            },
+        },
+        sharding::NotSharded,
         test_executor::run,
         test_fixture::{ipa::TestRawDataRecord, Reconstruct, Runner, TestWorld},
     };
 
+    fn input_row(
```

Comment:
a bit of bikeshedding, but this does generate an input row and expected result, so name can reflect that